controllers/suite_test.go

```diff
@@ -70,12 +75,20 @@ var _ = BeforeSuite(func() {
 	log = ctrl.Log.WithName("testdebug")
 
 	By("bootstrapping test environment")
+
+	// Envtest doesn't respect kustomization.yaml for CRDs, apply it ourselves
+	crdsTempfile, err := os.CreateTemp("", "crds*.yaml")
+	Expect(err).NotTo(HaveOccurred())
+	defer os.Remove(crdsTempfile.Name())
+	Expect(runKustomize(filepath.Join("..", "config", "crd"), crdsTempfile)).To(Succeed())
+	crdsTempfile.Close()
+
 	applicationServiceDepVersion := "v0.0.0-20230717184417-67d31a01a776"
 	applicationApiDepVersion := "v0.0.0-20230704143842-035c661f115f"
 	spiApiDepVersion := "v0.2023.22-0.20230731122342-dfc17adc9829"
 	testEnv = &envtest.Environment{
 		CRDDirectoryPaths: []string{
-			filepath.Join("..", "config", "crd", "bases"),
+			crdsTempfile.Name(),
```

Comment:
What if `bases` would have another CRD?

Reply:
`kubectl kustomize` will return a yaml file that includes all of them, separated by `---`

Envtest can handle this: https://github.com/kubernetes-sigs/controller-runtime/blob/edb7fe9c520ef2eb5e7d54ded35734bd9a10d71d/pkg/envtest/crd.go#L406 (also tested locally)

Reply:
well, there are now two yamls in bases, and 1 kustomize which changes both of them

but if there will be new yaml in bases, without change in kustomize I'd say that :
filepath.Join("..", "config", "crd", "bases"),

would be still required

so I'd just keep it there

Reply:
By my understanding, if a second CRD were added we would want to add it to the kustomization.yaml `resources` anyway. The `make install` command step makes the same assumption - `kustomize build config/crd` will return all the CRDs
```make
.PHONY: install
install: manifests kustomize ## Install CRDs into the K8s cluster specified in ~/.kube/config.
    $(KUSTOMIZE) build config/crd | kubectl apply -f - || true
```